### ISSUE TITLE
Integration with rsyslog

### ISSUE THREAD

**Issue by micobarac** [whyscream/postfix-grok-patterns#152]
With rsyslog `json template`:

```
template(name="json-template"
  type="list") {
    constant(value="{")
      constant(value="\"@timestamp\":\"")     property(name="timereported" dateFormat="rfc3339")
      constant(value="\",\"@version\":\"1")
      constant(value="\",\"message\":\"")     property(name="msg" format="json")
      constant(value="\",\"sysloghost\":\"")  property(name="hostname")
      constant(value="\",\"severity\":\"")    property(name="syslogseverity-text")
      constant(value="\",\"facility\":\"")    property(name="syslogfacility-text")
      constant(value="\",\"programname\":\"") property(name="programname")
      constant(value="\",\"procid\":\"")      property(name="procid")
    constant(value="\"}\n")
}
```

How to get `postfix/smtpd` from rsyslog as `program`? `tag` you mention in the docs doesn't exist in rsyslog properties.

**Comment by whyscream** [comment]
I don't have any experience with rsyslog json template, so maybe you should ask around in a forum with some people well-versed in those techniques. But  based on your variable naming, i'd say that the value `postfix/smtpd` corresponds with `programname`.

**Comment by micobarac** [comment]
`programname` gets only `postfix` value. I don't know how to append `smtpd` to it.

**Comment by whyscream** [comment]
I don't know how the value for the `programname` key is extracted from the original postfix logging. Maybe it's a regular expression that only detects `a-z0-9` values, and ignores the `/` character (and everything following it)?

As I said: I have no experience with rsyslog so I can't really comment on that: you need to find a place where people can comment on that part of your setup (which you didn't even describe in this issue).

**Comment by micobarac** [comment]
I think I found the source of the problem.

```
parser.permitSlashInProgramName [on/off] available 8.25.0+

Default: off

This controls whether slashes in the “programname” property (the static part of the tag) are permitted or not. By default this is not permitted, but some Linux tools (including most importantly the journal) store slashes as part of the program name inside the syslogtag. In those cases, the programname is truncated at the first slash.

In other words, if the setting is off, a value of app/foo[1234] in the tag will result in a programname of app, and if an application stores an absolute path name like /app/foo[1234], the programname property will be empty (“”). If set to on, a syslogtag of /app/foo[1234] will result in a programname value of /app/foo and a syslogtag of app/foo[1234] will result in a programname value of app/foo.
```

I have a slightly older version of rsyslog 8.24.0-52.el7_8.2, which doesn't support slashes in program names.
